public/app/features/dashboard/state/getPanelOptionsWithDefaults.test.ts

```diff
@@ -0,0 +1,390 @@
+      const result = runScenario({
+        plugin: getPanelPlugin({ id: 'graph' }),
+        options: {},
+        defaults: {},
+        overrides: [],
+      });
+
+      expect(result).toMatchInlineSnapshot(`
+        Object {
+          "fieldConfig": Object {
+            "defaults": Object {
+              "custom": Object {},
+            },
+            "overrides": Array [],
+          },
+          "options": Object {},
+        }
+      `);
+    });
+  });
+
+  describe('When current options are emtpy', () => {
+    it('Should set defaults', () => {
+      const result = getPanelOptionsWithDefaults({
+        plugin: pluginA,
+        currentOptions: {},
+        currentFieldConfig: {
+          defaults: {},
+          overrides: [],
+        },
+      });
+
+      expect(result).toMatchInlineSnapshot(`
+        Object {
+          "fieldConfig": Object {
+            "defaults": Object {
+              "custom": Object {
+                "hideLines": false,
+              },
+              "thresholds": Object {
+                "mode": "absolute",
+                "steps": Array [
+                  Object {
+                    "color": "green",
+                    "value": -Infinity,
+                  },
+                  Object {
+                    "color": "red",
+                    "value": 80,
+                  },
+                ],
+              },
+            },
+            "overrides": Array [],
+          },
+          "options": Object {
+            "name": "hello",
+            "number": 10,
+            "showThresholds": true,
+          },
+        }
+      `);
+    });
+  });
+
+  describe('When there are current options and overrides', () => {
+    it('Should set defaults', () => {
+      const result = getPanelOptionsWithDefaults({
+        plugin: pluginA,
+        currentOptions: {
+          number: 20,
+          showThresholds: false,
+        },
+        currentFieldConfig: {
+          defaults: {
+            unit: 'bytes',
+            decimals: 2,
+          },
+          overrides: [],
+        },
+      });
+
+      expect(result).toMatchInlineSnapshot(`
+        Object {
+          "fieldConfig": Object {
+            "defaults": Object {
+              "custom": Object {
+                "hideLines": false,
+              },
+              "decimals": 2,
+              "thresholds": Object {
+                "mode": "absolute",
+                "steps": Array [
+                  Object {
+                    "color": "green",
+                    "value": -Infinity,
+                  },
+                  Object {
+                    "color": "red",
+                    "value": 80,
+                  },
+                ],
+              },
+              "unit": "bytes",
+            },
+            "overrides": Array [],
+          },
+          "options": Object {
+            "name": "hello",
+            "number": 20,
+            "showThresholds": false,
```

Comment:
Ok, so the options that are currently set are preserved and the rest is set to defaults?

Reply:
yes